### PR TITLE
ARROW-3603: [Gandiva][C++] Support building with ARROW_BOOST_VENDORED=ON

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

**Review by pravindra — APPROVED** [review]
lgtm

File: cpp/cmake_modules/GandivaBuildUtils.cmake [hunk 1]
```diff
@@ -91,7 +90,7 @@ function(add_gandiva_unit_test REL_TEST_NAME)
     ${CMAKE_SOURCE_DIR}/src
   )
   target_link_libraries(${TEST_NAME}
-    PRIVATE arrow_shared ${GANDIVA_TEST_LINK_LIBS} Boost::boost
+    PRIVATE arrow_shared ${GANDIVA_TEST_LINK_LIBS}
   )
   add_test(NAME ${TEST_NAME} COMMAND ${TEST_NAME})
   set_property(TEST ${TEST_NAME} PROPERTY LABELS gandiva,unittest ${TEST_NAME})
```
Comment (wesm) [inline cpp/cmake_modules/GandivaBuildUtils.cmake]: We should endeavor to remove this file (or merge with `gandiva/CMakeLists.txt`) before the next release

**Comment by wesm** [comment]
+1

File: cpp/cmake_modules/GandivaBuildUtils.cmake (hunk 1 above)
Comment (kou) [reply cpp/cmake_modules/GandivaBuildUtils.cmake]: OK.

**Pull request closed without merging**